### PR TITLE
fix(wallet): update swap token text

### PR DESCRIPTION
**Pull request by Pavneet-Sing**
- update UI to add dex aggregator web link and dialog
- show initial value of asset in market price
- fix override build warning

Resolves https://github.com/brave/brave-browser/issues/20463

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Verify swap label per linked issue
- Verify "0x" link on Swap screen 
- Verify info dialog about aggregator on swap screen when info icon is touched
- Market price must contain value of selected asset

### CODE REVIEW

**Comment by SergeyZhukovsky** [comment]
I don't think we want to add a clickable element inside a clickable layout. How hard it to click on `0x` exactly? We have the link we want to open on a Swap screen. The issue is just to change text on a layout with buy/send/swap options. @jamesmudgett what text do you think is better there instead of what we have `Swap crypto assets with Brave DEX aggregator.`. @lukemulks proposes `Swap tokens to tokens`. I would better go with a simple `Swap crypto assets`.

**Comment by Pavneet-Sing** [comment]
> I don't think we want to add a clickable element inside a clickable layout. How hard it to click on `0x` exactly? We have the link we want to open on a Swap screen. The issue is just to change text on a layout with buy/send/swap options. @jamesmudgett what text do you think is better there instead of what we have `Swap crypto assets with Brave DEX aggregator.`. @lukemulks proposes `Swap tokens to tokens`. I would better go with a simple `Swap crypto assets`.

I just kept it same as desktop. It is easy to open the `0x` link. Now I am currently building it for device to ensure the same on device (will update this comment soon in case of any issue). For title, I can be concise or detailed with:
- `Swap crypto assets `or `Swap tokens` or `Swap tokens to tokens`
- `Swap tokens with other token` or maybe `Swap tokens with other tokens using 0x dex aggregator` (similar to old title)

**Comment by Pavneet-Sing** [comment]
> Token isn't correct there as you can swap ETH to something or something to ETH. ETH is considered a coin, not a token. Crypto assets would fit better there. Plus we are going to add more chains that support Swap where native assets could be coins as well.

Seems like both terms `tokens` and `assets` both can be used interchangeably 
[Metamask](https://metamask.io/swaps/) is using the term `tokens`.
Reference:
 `This means that crypto tokens, which are also called crypto assets, [represent a certain unit of value](https://www.investopedia.com/terms/c/crypto-token.asp#citation-3)`

**Comment by SergeyZhukovsky** [comment]
There is probably not hard rule how to call it, but going with assets we don't wonder do we call it right or not IMHO. But please check with @jamesmudgett on final text.

**Comment by Pavneet-Sing** [comment]
> There is probably no hard rule how to call it, but going with assets we don't wonder do we call it right or not IMHO. But please check with @jamesmudgett on final text.

Assets sounds more comprehensive, BTW, it's working fine on mobile considering the size of the text `0x`, rest gonna discuss with @jamesmudgett.

File: android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java [hunk 1]
```diff
@@ -1395,6 +1395,7 @@ public void onClick(View v) {
                     popup.getMenuInflater().inflate(
                             R.menu.monthly_contribution_popup_menu, popup.getMenu());
                     popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
+                        @Override
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java:1398]: thanks for adding that as it was annoying :)

File: browser/ui/android/strings/android_brave_strings.grd [hunk 2]
```diff
@@ -2679,6 +2685,9 @@ If you don't accept this request, VPN will not reconnect and your internet conne
     <message name="IDS_WALLET_DEX_AGGREGATOR" desc="Brave wallet msg about using 0x for swap">
       Brave uses 0x as a DEX aggregator.
     </message>
+    <message name="IDS_SWAP_DEX_AGGREGATOR_NAME" desc="Brave wallet swap dex aggregator source name">
+      0x
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:2689]: Please add `translateable="false"` as we don't want 0x to be translated.

Comment (Pavneet-Sing) [reply browser/ui/android/strings/android_brave_strings.grd:2689]: Got it, this is being used with [indexOf to find the position](https://github.com/brave/brave-core/pull/11978/files#diff-30eae7637d039986bb60e7d56d19a411894f2bce4725def3d4dd0945ea5edc72R1345) in `IDS_WALLET_DEX_AGGREGATOR` so do I need to wrap `0x` in `ph` tag as well, like 
` Brave uses <ph>0x</ph> as a DEX aggregator.`? or not

Comment (SergeyZhukovsky) [reply browser/ui/android/strings/android_brave_strings.grd:2689]: yeah, let's do how it looks like on desktop

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Pull request merged**